src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/AuthDAO.scala

```diff
@@ -0,0 +1,11 @@
+package org.broadinstitute.dsde.rawls.dataaccess
+
+import org.broadinstitute.dsde.rawls.model.{RawlsGroupRef, WorkspaceName, RawlsGroup, RawlsUser}
+
+trait AuthDAO {
+  def saveUser(rawlsUser: RawlsUser, txn: RawlsTransaction): RawlsUser
+
+  def saveGroup(rawlsGroup: RawlsGroup, txn: RawlsTransaction): RawlsGroup
+
+  def saveWorkspaceAccessGroups(workspaceName: WorkspaceName, userEmail: String, txn: RawlsTransaction): Map[String, RawlsGroupRef]
+}
```

Comment:
Removing the WorkspaceAccessGroups should be done as part of https://broadinstitute.atlassian.net/browse/DSDEEPB-1800, and general Group in https://broadinstitute.atlassian.net/browse/DSDEEPB-1803.  Not sure about User removal.